### PR TITLE
Sort filterable list by active items

### PR DESCRIPTION
**Pull request by RyTechro**
# Changes proposed in this pull request:
- Sorting filterable items by active items. In practice this will for instance sort the fuel page of the restaurant or the baker to show the enabled items at the top, this gives a quick overview of what items are active.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java [hunk 1]
```diff
@@ -185,6 +185,17 @@ private void updateResources()
         allItems.clear();
         allItems.addAll(getBlockList(filterPredicate));
         allItems.addAll(getExceptions().stream().filter(storage -> filterPredicate.test(storage.getItemStack())).collect(Collectors.toList()));
+
+        allItems.sort((o1, o2) -> {
+            if(!building.isAllowedItem(id, new ItemStorage(o1.getItemStack())) && building.isAllowedItem(id, new ItemStorage(o2.getItemStack()))){
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java]: should probably store the isAllowedItem check in a boolean value to not query it twice in the else if case

Comment (RyTechro) [reply src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java]: Agreed

File: src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java [hunk 2]
```diff
@@ -187,9 +187,13 @@ private void updateResources()
...
+
+            boolean o1Allowed = building.isAllowedItem(id, new ItemStorage(o1.getItemStack()));
+            boolean o2Allowed = building.isAllowedItem(id, new ItemStorage(o2.getItemStack()));
+
+            if(!o1Allowed && o2Allowed){
                 return 1;
-            }else if(building.isAllowedItem(id, new ItemStorage(o1.getItemStack())) && !building.isAllowedItem(id, new ItemStorage(o2.getItemStack()))){
+            }else if(o1Allowed && !o2Allowed){
                 return -1;
             }else{
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/client/gui/ViewFilterableList.java]: Just missing formatting now. We format if/else in
{

}

**Comment by Raycoms** [comment]
> [CLA assistant check](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689) before we can accept your contribution.
> 
> **TechRo** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ldtteam/minecolonies?pullRequest=6689) it.

Also needing this

**Comment by RyTechro** [comment]
> > [CLA assistant check](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689)
> > Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689) before we can accept your contribution.
> > **TechRo** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ldtteam/minecolonies?pullRequest=6689) it.
> 
> Also needing this

Can't seem to make this work, if I clicked the details link it said that I've signed the SLA and now it just redirects back to this issue. I have signed the SLA and do agree with it.

**Comment by Raycoms** [comment]
> > > [CLA assistant check](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689)
> > > Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689) before we can accept your contribution.
> > > **TechRo** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> > > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ldtteam/minecolonies?pullRequest=6689) it.
> > 
> > 
> > Also needing this
> 
> Can't seem to make this work, if I clicked the details link it said that I've signed the SLA and now it just redirects back to this issue. I have signed the SLA and do agree with it.

Ah, it's because the commits were made with Techro and you signed it as RyTechro

**Comment by RyTechro** [comment]
> > > > [CLA assistant check](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689)
> > > > Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ldtteam/minecolonies?pullRequest=6689) before we can accept your contribution.
> > > > **TechRo** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> > > > You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ldtteam/minecolonies?pullRequest=6689) it.
> > > 
> > > 
> > > Also needing this
> > 
> > 
> > Can't seem to make this work, if I clicked the details link it said that I've signed the SLA and now it just redirects back to this issue. I have signed the SLA and do agree with it.
> 
> Ah, it's because the commits were made with Techro and you signed it as RyTechro

I see, my local config is messed up. Let me see if I can fix it

**Comment by RyTechro** [comment]
Opened a new PR to fix the CLA issue

**Pull request closed without merging**